### DISCUSSION TITLE
video player in web app

### DISCUSSION THREAD

**Discussion by 6jlarogap** [revenkroz/telegram-web-app-bot-example#9] [Q&A]
Have a good day!
Is it possible for bot in responce to message with video - open and play this video in webapp - with custom buttons with actions like voting after video?

Or, is it possible for bot to generate link to video file - that could be opened in any external html5 browser page?

Thanks!

**Comment by revenkroz** [comment]
Hello!

Unfortunately, I may not have understood the question very well. 

Do you want the bot to send a button that opens a WebApp and starts a video with the right buttons and everything?

**Comment by 6jlarogap** [comment]
Юзер отправляет боту видео - бот получает у видео урл, генерит и записывает в бд токен логина юзера, и возвращает ссылку или кнопку на вэбапп с html5 проигрывателем и кнопками голосования - кнопки нажимаются _много_ раз по мере просмотра видео - события нажатия включая время просмотра видео в момент нажатия - отправляются во внешний апи* для суммирования и построения графика.

![IMG_20230219_092257](https://user-images.githubusercontent.com/62824780/219932561-33bd640d-42fd-40ef-b689-ae7a1a585fcf.jpg)

На картинке - сверху - играющий хтмл5 проигрыватель с текущим временем воспроизведения - под ним - кнопки голосования - с отправкой во внешний апи* - юзертокен|видеоид**|кнопка|время;
Далее - график получаемый из апи* - с суммой голосов всех пользователей;
Журнал голосов текущего пользователя с возможностью отмены;

*Внешний апи имеется - нужен только фронт;
**Для Видеоид похоже нельзя использовать ссылку на видео так как телега вроде меняет их постоянно - вероятно можно передавать сюда ид сообщения с видео - при пересылке этого сообщения между пользователями и боту - он вроде должен сохраниться.

Вероятно можно сделать такую страницу и без вэбапп - вне телеграма - и переходить на неё по ссылке от бота с токеном и видеоид в параметрах - вэбапп ведь везде работает где есть телеграм?
